docs/DeveloperGuide.md

```diff
@@ -1,34 +1,600 @@
+
+    delete appt 1 
+    
+will delete the appointment with appointment ID `1`, if found. 
+
+**Implementation** 
+
+The following steps explains the sequence of events: 
+
+1. The user enters `delete appt 1`. 
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`.
+
+3. `AppointmentParser#parse()` will then create an object `DeleteAppointmentCommand`. 
+
+4. `HappyPills` then calls `DeleteAppointmentCommand.execute()` to execute the command. 
+
+5. In `DeleteAppointmentCommand.execute()`, the respective patient is deleted and the display message is returned. 
+
+#### 4.3.4. Mark Appointment as Done 
+
+#### 4.3.5. List Appointments 
+
+The user is able to get a list of all the appointments currently in the program.
+The command: 
+    
+    list appt
+    
+will list all the appointments in the `AppointmentMap`. 
+
+**Implementation** 
+
+![List Appointment Sequence Diagram](images/ListAppointmentSequenceDiagram.jpg)
+
+The following steps explains the sequence of events: 
+
+1. The user enters `list appt`.
+
+2. `HappyPills` calls `Parser#parse()` which then calls `AppointmentParser#parse()`.
+
+3. `AppointmentParser#parse()` will create an object `ListAppointmentCommand`.
+
+4. `HappyPills` then calls `ListAppointmentCommand.execute()` to execute the method. 
+
+5. In `ListAppointmentCommand.execute()`, it calls `TextUi.getAppointmentList()` which is then returned as the display message. 
+
+#### 4.3.6. Find Appointments of Patient
+
+The user is able to find all the appointments a specific patient has.
+The command: 
+    
+    find appt S7777777Z
+    
+will list all the appointments that the patient with NRIC S7777777Z has. 
+
+**Implementation**
+
+![Find Appointment Sequence Diagram](images/FindAppointmentSequenceDiagram.jpg)
+
+
+### 4.4. Storage
+
+This is an internal feature of the program, implemented to allow users to recover information even after HappyPills is 
+closed in the terminal. This is achieved by storing all relevant information in a text file using a structured format.
+
+**Implementation** 
+
+ The current methods implemented in this class and a brief description of each method:
+ - `writeAllToFile` — writes the entire list of item to the specified text file.
+ - `addSingleItemToFile` — adds a new item as a single string to the end of specified text file.
+ - `loadPatientFromFile` — access the patient file and retrieve all information in the file as strings.
+ - `parsePatientFileContent` — process line-by-line to create a patient object and add to the shared patient map.
+ - `loadAppointmentFromFile` — access the patient file and retrieve all information in the file as strings.
+ - `parseAppointmentFileContent` — process line-by-line to create an appointment object and add to the 
+                                   shared appointment map.
+ 
+ The following diagram shows how each command interacts with the other classes.
+ 
+ *writeAllToFile*
```

Comment:
If this is meant to be a header, try to make it more obvious? It is easy to overlook because it is the same size and design of the surrounding text. Similarly to the following headers with the same design